ios/CustomerioReactnative.swift

```diff
@@ -2,7 +2,17 @@ import Foundation
 import CioTracking
 import Common
 import CioMessagingInApp
+import UserNotifications
 
+enum PushPermissionStatus : String {
```

Comment:
I believe that the purpose of this enum is to convert the Swift data type `UNAuthorizationStatus` to a String. 

I do want to propose the idea of simplifying the push permission status options. RN customers may not care about all of these results and they would probably enjoy a permission status that is similar to Android if possible. 

What if we were to simplify this enum to be 3 cases:
* Granted
* Denied
* Not asked 

Sure, some of these `UNAuthorizationStatus` values such as `ephemeral` mean that the push permission is granted, but only for a limited time. But to a RN customer, they may not care about that fine of detail.

Reply:
With react native getting competitive with native platforms, I think a very few but there could be customers using `ephemeral`. I do not mind removing it though but I think adding it also doesn't do any harm. 🤔

Reply:
We could make the `CustomerIOTracking.getPushPermissionStatus` function return a typescript object. 

One property of the object gives a simplified status, one property gives a raw status. 

```ts
let pushStatusResult = await CustomerIOTracking.getPushPermissionStatus()

pushStatusResult.status // an enum that's one of 3 options: granted, denied, not asked
pushStatusResult.rawOSStatus // a String that returns the raw result from the OS. For iOS, that would mean we return a string mapping to `UNAuthorizationStatus` raw value. 
```

Reply:
Giving more thoughts to this, I think simplifying to just 3 cases would be ideal. I am implementing the same so that we remain consistent across platforms.